leader_test.go

```diff
@@ -488,44 +491,70 @@ func Test_namespacesList(t *testing.T) {
 }
 
 func Test_getServiceInstances(t *testing.T) {
-	ts := httptest.NewServer(http.HandlerFunc(
-		func(w http.ResponseWriter, r *http.Request) {
-			uri := r.RequestURI
-			switch { // ignore anything that doesn't require a return body
-			case strings.Contains(uri, "status/leader"):
-				fmt.Fprint(w, `"127.0.0.1"`)
-			case strings.Contains(uri, "namespace"):
-				fmt.Fprint(w, namespacesJSON)
-			case strings.Contains(uri, "health/service"):
-				fmt.Fprint(w, healthserviceJSON)
+	partitionQueryParamKey := "partition"
+	// parameterized test
+	cases := []struct {
+		name, partition string
+	}{
+		{"No partition", ""},
+		{"default partition", "default"},
+		{"admin partition", "admin"},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(
+				func(w http.ResponseWriter, r *http.Request) {
+					uri := r.RequestURI
+					switch { // ignore anything that doesn't require a return body
+					case strings.Contains(uri, "status/leader"):
+						assert.Equal(t, tc.partition, r.URL.Query().Get(partitionQueryParamKey))
+						fmt.Fprint(w, `"127.0.0.1"`)
+					case strings.Contains(uri, "namespace"):
+						assert.Equal(t, tc.partition, r.URL.Query().Get(partitionQueryParamKey))
+						fmt.Fprint(w, namespacesJSON)
+					case strings.Contains(uri, "health/service"):
+						assert.Equal(t, tc.partition, r.URL.Query().Get(partitionQueryParamKey))
+						fmt.Fprint(w, healthserviceJSON)
+					}
```

Comment:
minor, but could we add a default case here that errors if an unexpected endpoint is called so we can be sure that we're catching all the cases?